src/generator/passes.jl

```diff
@@ -756,6 +756,51 @@ function (x::TweakMutability)(dag::ExprDAG, options::Dict)
     return dag
 end
 
+"""
+    AddFPtrMethods <: AbstractPass
+This pass adds a method definition for each function prototype method which `ccall`s into a library.
+
+The generated method allows the use of a function pointer to `ccall` into directly, instead
+of relying on a library to give the pointer via a symbol look up. This is useful for libraries that
+use runtime loaders to dynamically resolve function pointers for API calls.
+"""
+struct AddFPtrMethods <: AbstractPass end
+
+function (::AddFPtrMethods)(dag::ExprDAG, options::Dict)
+    codegen_options = get(options, "codegen", Dict())
+    use_ccall_macro = get(codegen_options, "use_ccall_macro", false)
+    for node in dag.nodes
+        node.type isa FunctionProto || continue
+        for _ex in copy(node.exprs)
+            ex = deepcopy(_ex)
+            Base.is_expr(ex, :function) || continue
```

Comment:
Not sure if there would actually be expressions for nodes of type `FunctionProto` that are not functions.